### PR TITLE
Add public external IDs to tournament and matches

### PR DESCRIPTION
**Pull request by n8kim1**
When a tournament is run, this will create external IDs for all the public stuff. This should be enough to run a tournament and be able to publish things later.

If the model looks good I'll make migration then, don't wanna make-rollback-remake

Does half of #574

### CODE REVIEW

File: backend/siarnaq/api/episodes/models.py [hunk 1]
```diff
@@ -302,10 +302,17 @@ def initialize(self):
...
-
         bracket.bulk_add_teams(self, teams, is_private=True)
+
         bracket.start_tournament(self, is_private=True)
 
+        # Also create a bracket for public display
+        # Make sure teams stay the same for consistency!
+        bracket.create_tournament(self, is_private=False)
+        bracket.bulk_add_teams(self, teams, is_private=False)
+        bracket.start_tournament(self, is_private=False)
```
Comment (j-mao) [inline backend/siarnaq/api/episodes/models.py:313]: Can this just be

```python
teams = ...

for is_private in [False, True]:
    bracket.create_tournament(self, is_private=is_private)
    bracket.bulk_add_teams(self, teams, is_private=is_private)
    bracket.start_tournament(self, is_private=is_private)
```

File: backend/siarnaq/bracket/challonge.py [hunk 2]
```diff
@@ -141,17 +141,63 @@ def get_round_indexes(tournament: Tournament, *, is_private: bool):
...
+    # Compute a correspondence between the two.
+    # Currently, we use the fact that IDs are in the same order for all tournaments.
+    # You could also use other fields, such as Challonge's `identifier` or
+    # `suggested_play_order` for matches; these are equal across brackets.
+    # Similarly you could use `seed` for teams.
+    matches_private.sort(key=lambda i: i["id"])
+    matches_public.sort(key=lambda i: i["id"])
+    matches_private_to_public = dict()
+    for c1, c2 in zip(matches_private, matches_public):
+        matches_private_to_public[c1] = c2
```
Comment (j-mao) [inline backend/siarnaq/bracket/challonge.py]: `c1` is a dict?

File: backend/siarnaq/bracket/challonge.py [hunk 3]
```diff
@@ -141,17 +141,63 @@ def get_round_indexes(tournament: Tournament, *, is_private: bool):
     return round_indexes
 
 
-def get_match_and_participant_objects_for_round(
-    round: TournamentRound, *, is_private: bool
-):
-    tournament_data = get_tournament_data(round.tournament, is_private=is_private)
-    # Derive match dicts/JSON objects (that Challonge gives us) of this round
+def _pair_public_private_challonge_ids(tournament: Tournament):
```
Comment (j-mao) [inline backend/siarnaq/bracket/challonge.py:144]: I feel like this function does the same thing twice: once on items with type `match` and once with `participant`… and the behaviour is identical for both cases.

Perhaps accept the type as a function argument?

File: backend/siarnaq/bracket/challonge.py [hunk 4]
```diff
@@ -198,6 +255,9 @@ def get_match_and_participant_objects_for_round(
             challonge_participant_id = challonge_match["relationships"][
```
Comment (j-mao) [inline backend/siarnaq/bracket/challonge.py:259]: Rename to include `private` in name.

**Review by j-mao — COMMENTED** [review]
lgtm, hope it works in staging

File: backend/siarnaq/api/episodes/models.py [hunk 5]
```diff
@@ -298,7 +298,7 @@ def initialize(self):
         self.external_id_private = f"private_{self.external_id_public}_{key}"
 
         teams = self.get_eligible_teams()
-        for is_private in [False, True]:
+        for is_private in [True, False]:
```
Comment (j-mao) [inline backend/siarnaq/api/episodes/models.py:301]: Just wondering, was there a reason for swapping them around?

**Review by j-mao — APPROVED** [review]
Looks good to me

File: backend/siarnaq/api/episodes/models.py (hunk 5 above)
Comment (n8kim1) [reply backend/siarnaq/api/episodes/models.py:301]: the overall tour workflow is basically "do private first then public!". and same for code.
When I was testing, i was looking at IDs, seeing private IDs be _higher_ than public ones. This goes against "private first then public" and so it gave me a headache and i didn't like it

**Pull request merged**